### PR TITLE
feat: Add support for cylc configuration files

### PR DESCRIPTION
**Pull request by elliotfontaine**
> Cylc (pronounced silk) is a general purpose workflow engine that also manages cycling systems very efficiently. It is used in production weather, climate, and environmental forecasting on HPC, but is not specialized to those domains.

[Cylc](https://github.com/cylc/cylc-flow)

Configurations are the main filetype of Cylc, as they're used to define workflows (tasks and their dependencies).

### CODE REVIEW

File: runtime/queries/cylc/injections.scm
Comment (the-mikedavis) [inline runtime/queries/cylc/injections.scm:2]: I believe you need double backslashes here since the first is escaped:

```suggestion
  key: (key) @key (#match? @key "^script$|\\-script$|^script\\-")
```

but also I think it shouldn't be necessary to escape the `-`. IIRC it needs to be escaped when used in the middle of a character class `[a\\-z]` but not otherwise. I haven't verified this though so it's worth checking if I'm correct here

File: runtime/queries/cylc/injections.scm
Comment (the-mikedavis) [inline runtime/queries/cylc/injections.scm:14]:
```suggestion
    name: (_) @section_name (#eq? @section_name "environment")
```

File: runtime/queries/cylc/injections.scm
Comment (elliotfontaine) [reply runtime/queries/cylc/injections.scm]: I chose to remove the escaping entirely. As you said, it was unecessary.

File: runtime/queries/cylc/highlights.scm
Comment (the-mikedavis) [inline runtime/queries/cylc/highlights.scm:24]: there's a typo in these "ponctuation" => "punctation"

File: runtime/queries/cylc/highlights.scm
Comment (the-mikedavis) [inline runtime/queries/cylc/highlights.scm:12]:
```suggestion
  path: _ @string.special.path)
```

File: runtime/queries/cylc/highlights.scm
Comment (elliotfontaine) [reply runtime/queries/cylc/highlights.scm:24]: I was actually gonna change it to "p***u***nct***u***ation" but I can make it "punctation" if you insist 👽

btw, I'm adding support to Neovim, and they have a [`check-queries.lua`](https://github.com/nvim-treesitter/nvim-treesitter/blob/master/scripts/check-queries.lua) script to catch wrong capture names. Since Helix doesn't use Lua I guess it would have to be rewritten from scratch. 😕

**Review by the-mikedavis — APPROVED** [review]

**Comment by elliotfontaine** [comment]
Hi @the-mikedavis.
Should I squash the commits now or wait for the second review ?

**Comment by the-mikedavis** [comment]
We can squash as we merge so let's wait for another reviewer

**Comment by pascalkuthe** [comment]
Needs a rebase then we can merge

**Comment by elliotfontaine** [comment]
> Needs a rebase then we can merge

Should be good now.

**Review by pascalkuthe — APPROVED** [review]

**Review by the-mikedavis — APPROVED** [review]

**Pull request merged**
